### PR TITLE
Fix dynamic propertylookup

### PR DESCRIPTION
**Pull request by czosel**
There are probably many cases where this still breaks, but it might be an improvement anyway :smile: 
#112

### CODE REVIEW

**Comment by mgrip** [comment]
is it worth considering doing this from the parent level instead of the child?

ie when we print the property lookup, and `node.offset.kind !== "constref"`, then wrap in curly's? Might be simpler than trying to go the other way and trying to catch every edge case

**Comment by czosel** [comment]
@mgrip Of course, i can't believe i didn't notice that :sweat_smile:

File: tests/encapsed/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -23,6 +23,6 @@ $encapsHereDoc2 = <<<STR
 Hello foo $a bar
 new line
 STR;
-$encapsedOffset = $a->bar$b;
+$encapsedOffset = $a->{bar$b};
```
Comment (czosel) [inline tests/encapsed/__snapshots__/jsfmt.spec.js.snap]: Is `$a->bar$b` valid PHP?

Comment (nicoder) [reply tests/encapsed/__snapshots__/jsfmt.spec.js.snap]: I don't think so, it seems to be a parse error in PHP7

```
php > $a = new stdClass;
php > $a->foobar = 1;
php > $b = 'bar';
php > echo $a->{"foo$b"};
1
php > echo $a->foo$b;

Parse error: syntax error, unexpected '$b' (T_VARIABLE), expecting ',' or ';' in php shell code on line 1
php > echo $a->{foo$b};

Parse error: syntax error, unexpected '$b' (T_VARIABLE) in php shell code on line 1
```

Comment (czosel) [reply tests/encapsed/__snapshots__/jsfmt.spec.js.snap]: :+1: i'll remove it from the tests
